### PR TITLE
Fix Bug 1133095, Vitals page inconsistency

### PR DESCRIPTION
**Pull request by lferreirawork**
Bug 1133095:
- Sometimes it takes a while before the rollovers to activate the groups.
- Dots will get out of alignment (http://cl.ly/image/1K1t1e2E410p)
- Groups will stay expanded after rolling out (http://cl.ly/image/420y1f3L080U)

also:
- Background dots are not showing up after hovering a vitals group
- Issues are faded out when you move from Ecosystem to Vitals

### CODE REVIEW

File: src/scripts/experience/topic.js [hunk 1]
```diff
@@ -178,9 +182,9 @@ define([
             issue.tapS.remove(issue.topicTap);
         }
 
-        for(i = 0; i < this._fakes.length; i ++) {
-            this._fakes[i].explode();
-        }
+        // for(i = 0; i < this._fakes.length; i ++) {
+        //     this._fakes[i].explode();
+        // }
```
Comment (schalkneethling) [inline src/scripts/experience/topic.js]: If the above is not required, you can just delete it, no need to comment it out.

File: src/scripts/experience/topic.js [hunk 2]
```diff
@@ -223,55 +227,53 @@ define([
     * When hovering a topic. If force is true, ignore the check to see if this is
     * a valid mouseOver.
     */
-    Topic.prototype._mouseOver = function (force) {
+    Topic.prototype._mouseOver = function () {
+        // console.log('_mouseOver', this._index);
```
Comment (schalkneethling) [inline src/scripts/experience/topic.js]: Nit: remove console.log

File: src/scripts/experience/topic.js [hunk 3]
```diff
@@ -223,55 +227,53 @@ define([
...
-            posX = this.elm.x + this._linearArea.x;
-            posY = this.elm.y + this._linearArea.y;
+            posX = this.elm.x + this._linearPos.x;
+            posY = this.elm.y + this._linearPos.y;
             posY += (((this._linearDist * i) - this._linearDist * this._issues.length / 2));
             issue.moveTo(Math.round(posX), Math.round(posY), null, {alpha: 1});
         }
 
-        // hide fakes
-        for(i = 0; i < this._fakes.length; i ++) {
```
Comment (schalkneethling) [inline src/scripts/experience/topic.js]: Can rewrite this as:

    for(i = 0, l = this._fakes.length; i < l; i++) {

File: src/scripts/experience/topic.js [hunk 4]
```diff
@@ -223,55 +227,53 @@ define([
...
-        // this._linearArea.mouseout = this._linearArea.touchend = this._mouseOut.bind(this);
-
-        setTimeout(function () {
-            this.isOver = true;
-        }.bind(this), 300);
 
-        this.elm.removeChild(this._topicArea);
-        this.elm.addChild(this._linearArea);
+        // hide fakes
+        for(i = 0; i < this._fakes.length; i ++) {
```
Comment (schalkneethling) [inline src/scripts/experience/topic.js]: Same as above here.

File: src/scripts/experience/topic.js [hunk 5]
```diff
@@ -223,55 +227,53 @@ define([
...
-                if (!this._isMouseOver()) {
-                    this._mouseOut();
-                }
-            }.bind(this), 100);
+        if (this._mouseTimeout) {
+            clearTimeout(this._mouseTimeout);
         }
+        // update state when the issues stop moving
+        this._mouseTimeout = setTimeout(function () {
+            // console.log('_mouseOver end', this._index, this._isMouseOver());
```
Comment (schalkneethling) [inline src/scripts/experience/topic.js]: Nit: remove console.log

File: src/scripts/experience/topic.js [hunk 6]
```diff
@@ -281,8 +283,10 @@ define([
     * if this is a valid mouseOut.
     */
     Topic.prototype._mouseOut = function (force) {
+        // console.log('_mouseOut', this._index);
```
Comment (schalkneethling) [inline src/scripts/experience/topic.js]: Nit: and here

File: src/scripts/experience/topic.js [hunk 7]
```diff
@@ -281,8 +283,10 @@ define([
...
         // make sure it's really a mouse out
         if (force !== true && this._isMouseOver()) {
+            // console.log('force _mouseOver', this._index);
```
Comment (schalkneethling) [inline src/scripts/experience/topic.js]: Nit: and here

File: src/scripts/experience/topic.js [hunk 8]
```diff
@@ -303,59 +325,48 @@ define([
...
-        setTimeout(function () {
-            // if (Topic.SELECTED_TOPIC === this) {
-                this.elm.addChild(this._topicArea);
-                this.elm.removeChild(this._linearArea);
-                Topic.SELECTED_TOPIC = null;
-            // }
-        }.bind(this), 300);
+        // update state when the issues stop moving
+        this._mouseTimeout = setTimeout(function () {
+            // console.log('_mouseOut end', this._index, this._isMouseOver());
```
Comment (schalkneethling) [inline src/scripts/experience/topic.js]: Nit: and here

File: src/scripts/experience/topic.js [hunk 9]
```diff
@@ -303,59 +325,48 @@ define([
...
+        }.bind(this), 500);
 
         this.mouseOutS.dispatch(this);
     };
 
     Topic.prototype.toneDown = function () {
-        gs.TweenMax.to(this._topicTitle, 0.3, {alpha: 0.5});
-        gs.TweenMax.to(this._topicDesc, 0.3, {alpha: 0.5});
+        // gs.TweenMax.to(this._topicTitle, 0.3, {alpha: 0.5});
+        // gs.TweenMax.to(this._topicDesc, 0.3, {alpha: 0.5});
```
Comment (schalkneethling) [inline src/scripts/experience/topic.js]: remove instead of commenting out

File: src/scripts/experience/topic.js [hunk 10]
```diff
@@ -303,59 +325,48 @@ define([
...
-        gs.TweenMax.to(this._topicDesc, 0.3, {alpha: 1});
+        // gs.TweenMax.to(this._topicTitle, 0.3, {alpha: 1});
+        // gs.TweenMax.to(this._topicDesc, 0.3, {alpha: 1});
 
-        for(var i = 0; i < this._issues.length; i ++) {
-            gs.TweenMax.to(this._issues[i].elm, 1, {alpha: 1});
-        }
+        // for(var i = 0; i < this._issues.length; i ++) {
+        //     gs.TweenMax.to(this._issues[i].elm, 1, {alpha: 1});
+        // }
```
Comment (schalkneethling) [inline src/scripts/experience/topic.js]: Same for the above three

File: src/scripts/experience/topic.js [hunk 11]
```diff
@@ -303,59 +325,48 @@ define([
...
     };
 
     Topic.prototype.moveTo = function (position) {
+        // console.log('moveTo', this._index);
```
Comment (schalkneethling) [inline src/scripts/experience/topic.js]: remove

File: src/scripts/experience/topics-mode.js [hunk 12]
```diff
@@ -153,6 +153,14 @@ define([
     };
 
     TopicsMode.prototype._onMouseOver = function (selectedTopic) {
+        // if (this.selectedTopic) {
+        //     console.log('_onMouseOver', this.selectedTopic._index, selectedTopic._index);
+        // }
```
Comment (schalkneethling) [inline src/scripts/experience/topics-mode.js]: remove above

File: src/scripts/experience/topics-mode.js [hunk 13]
```diff
@@ -167,6 +175,9 @@ define([
     };
 
     TopicsMode.prototype._onMouseOut = function (selectedTopic) {
+        // if (this.selectedTopic) {
+        //     console.log('_onMouseOut', this.selectedTopic._index, selectedTopic._index);
+        // }
```
Comment (schalkneethling) [inline src/scripts/experience/topics-mode.js]: remove

File: src/scripts/experience/topics-mode.js [hunk 14]
```diff
@@ -233,7 +244,7 @@ define([
             this._canvas.swipeRightS.add(this._swipeToPrevTopicBind);
         }
 
-        this._onResize();
+        // this._onResize();
```
Comment (schalkneethling) [inline src/scripts/experience/topics-mode.js]: and here

**Comment by schalkneethling** [comment]
Some nits, but lgtm :+1: Please just squash the commits into one, once the above items has been resolved and we should be good to merge. Thanks!

**Comment by lferreirawork** [comment]
Removed unnecessary pieces of code. I didn't change the for loops though, so they are consistent across the project (not storing array length anywhere else), hope that's ok!

**Comment by schalkneethling** [comment]
np, I have actually been changing them in other places but, not a big deal. I will handle those. Thanks for the PR! \o/ r+

**Pull request merged**